lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -614,6 +616,60 @@ emptyWalletWith ctx (name, passphrase, addrPoolGap) = do
     expectResponseCode @IO HTTP.status202 r
     return (getFromResponse id r)
 
+-- | Prepare Wallet with utxo of size n (each utxo having = amt) and payload for Tx
+-- with n number of inputs (single input = amtToSend) to be sent from that wallet
+fixtureNInputs
+    :: forall t. (EncodeAddress t, DecodeAddress t)
+    => Context t
+    -> (Int, Natural)
+    -> Natural
+    -> IO (ApiWallet, ApiWallet, Payload)
+fixtureNInputs ctx (n, amt) amtToSend = do
+    wSrc <- fixtureWalletWith ctx (replicate n amt)
+    wDest <- emptyWallet ctx
+    addrs <- listAddresses ctx wDest
+
+    -- each amtToSend is going to the same address
+    let addrIds = view #id <$> take n addrs
+    let addrIdRepl = replicate n addrIds !! 1
+    let payments = flip map addrIdRepl $ \addr -> [aesonQQ|{
+            "address": #{addr},
+            "amount": {
+                "quantity": #{amtToSend},
+                "unit": "lovelace"
+            }
+        }|]
+    let payload = Json [aesonQQ|{
+            "payments": #{payments :: [Value]},
+            "passphrase": "Secure Passphrase"
+        }|]
+    return (wSrc, wDest, payload)
+
+-- | Prepare Wallet with utxo of size n (each output = amt) and payload for Tx
+-- to be send with given amtToSend (single input)
+fixtureMaxTxSize
```

Comment:
can't we marry this fixture with `fixtureNInputs` and create one ? They seem to share a lot code

Reply:
or at least use `payments` and `payload` The code here seems to be repeated in several places, I see it also in `fixtureWalletWith`

Reply:
merged `fixtureNInputs` and `fixtureMaxTxSize` together.